src/coreclr/src/pal/src/loader/module.cpp

```diff
@@ -1435,6 +1436,36 @@ static LPWSTR LOADGetModuleFileName(MODSTRUCT *module)
     return module->lib_name;
 }
 
+static bool ShouldRedirectToCurrentLibrary(LPCSTR libraryNameOrPath)
+{
+    if (!g_running_in_exe)
+        return false;
+
+    // Getting nullptr as name indicates redirection to current library
+    if (libraryNameOrPath == nullptr)
+        return true;
+
+    const char* toRedirect[] = {
+        "System.Native.so",
```

Comment:
Is the plan to merge all of these libraries into the host with this change? This code assumes that, it was done in the previous experiment since we were trying to merge everything.

Reply:
I do not know yet. If they are always needed, we can. 
I just ported the changes from the prototype. I am not sure if this will stay.